model.py

```diff
@@ -0,0 +1,182 @@
+        # self.df_train = pd.read_csv(self.df_train_path)
+        self.preprocess = Preprocess(df_path=self.df_train_path)
+        self.df_train = self.preprocess.get_df()
+        self.survived = self.df_train['Survived']  # target variable
+        self.features = list(self.df_train.columns)
+        self.pipeline = Pipeline(steps=[], verbose=True)
+
+    def get_pipeline(self, model=None, model_name: str = 'model', train: bool = False) -> Pipeline:
+        """
+        Return a pipeline to preprocess data and bundle with a model.
+        Args:
+            model: scikit-learn instantiated model object, e.g. XGBClassifier (or scikit-learn model compatible
+            e.g. xgboost library)
+            model_name: model name to be in the pipeline
+            train: if used for model selection then we don't need to perform feature selection every as it is the same
+            regardless of what model is being used to classify
+
+        Returns:
+            Pipeline (object): Pipeline steps.
+        """
+        if train and os.path.exists(train_preprocessed_path):  # used for model selection
+            assert model is not None, 'model shouldn\'t be None for model selection run'
+            # function used for model selection during train
+            pipeline = Pipeline(steps=[('model', model)])
+            pipeline.set_output(transform="pandas")
+            return pipeline
+
+        # preprocess pipeline + feature selection + model to classify for test purposes
+        pipeline = self.preprocess.get_preprocess_pipeline()
+        pipeline.steps.append(['poly', PolynomialFeatures(include_bias=False, interaction_only=True)])
+
+        # dropping interaction features that are columns of zeros
+        pipeline.steps.append(['drop_constant_features', DropConstantFeatures()])
+        pipeline.steps.append(['feature_select',
+                               RFECV(estimator=LogisticRegression(), step=1, cv=5, scoring='roc_auc')])
+        pipeline.set_output(transform="pandas")
+
+        # running for creating preprocessed train dataset + feature selection
+        if train and model is None:
+            return pipeline
+
+        # running for testing model with preprocessing data for example running on test dataset
+        pipeline.steps.append([model_name, model])
+        return pipeline
+
+    def select_model(self, X: pd.DataFrame, y: pd.DataFrame) -> pd.DataFrame:
+        """Test a variety of classifiers and return their performance metrics on training data.
+            modified code from
+            https://practicaldatascience.co.uk/machine-learning/how-to-create-a-contractual-churn-model
+        Args:
+            X (object): Pandas dataframe containing X_train data.
+            y (object): Pandas dataframe containing y_train data.
+
+        Return:
+            df (object): Pandas dataframe containing model performance data.
+        """
+        max_pipeline_score = 0
+
+        classifiers = {}
```

Comment:
1. Instead of this, use a for loop on a dictionary with these values
2. You can initiate this classifiers dictionary outside of this function (Maybe at `__init__`)
3. Do you really need all  those classifiers?